package/MDAnalysis/analysis/hbonds/hbond_autocorrel.py

```diff
@@ -231,9 +291,14 @@ def __init__(self, universe,
 
         self.h = hydrogens
         self.a = acceptors
-        self.d = donors
+        if donors is not None:
+            self.d = donors
+        else:
+            # with fingers.crossed():
+            self.d = find_hydrogen_donors(hydrogens)
```

Comment:
Can this fail "properly"? 

Is it safe to do this as a default?

Reply:
reverted this, so the responsibility is always to check donor AG rather than have class silently do it